lib/ecto/query/builder/from.ex

```diff
@@ -145,6 +140,37 @@ defmodule Ecto.Query.Builder.From do
   def prefix!(prefix) when is_binary(prefix) or is_nil(prefix), do: prefix
   def prefix!(prefix), do: raise("`prefix` must be a string, got: #{inspect(prefix)}")
 
+  @doc """
+  Validates hints at compile time and runtime
+  """
+  def hint!(hint) when is_binary(hint), do: hint
+
+  def hint!({:unsafe_fragment, _, [fragment]}) do
+    case fragment do
+      {:^, _, [value]} ->
+        quote do
+          checked = Ecto.Query.Builder.From.hint!(unquote(value))
+          {:unsafe_fragment, checked}
```

Comment:
I think we can actually unpack it here:

```suggestion
          Ecto.Query.Builder.From.hint!(unquote(value))
```

This way we don't need to change the adapters. WDYT?

Reply:
Didn't realise adapter would need changes. If yes, it would be relevant to notify people from ecto_ch as they use the hints feature.

Reply:
Yeah at first I was thinking to keep it for inspect but maybe it is not worth it. Do you think it’s worth it for inspect?

Reply:
The clickhouse people need to be notified either way though because the kw version is deprecated now

Reply:
Yeah, i don't think it is worth it. :)